src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,136 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = getNode(index);
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        if (currentNode.prev != null) {
+            currentNode.prev.next = newNode;
+        } else {
+            head = newNode;
+        }
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldItem = node.item;
+        node.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removeNode = getNode(index);
+        size--;
+        if (removeNode == head) {
+            head = removeNode.next;
+        } else {
+            removeNode.prev.next = removeNode.next;
+        }
+        if (removeNode == tail) {
+            tail = removeNode.prev;
+        } else {
+            removeNode.next.prev = removeNode.prev;
+        }
+        return removeNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (object == currentNode.item || (object != null && object.equals(currentNode.item))) {
+                if (currentNode == head) {
+                    head = currentNode.next;
+                } else {
+                    currentNode.prev.next = currentNode.next;
+                }
+                if (currentNode == tail) {
+                    tail = currentNode.prev;
+                } else {
+                    currentNode.next.prev = currentNode.prev;
+                }
+                size--;
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
```

Comment:
In these two methods we need to create method unlink(Node node) according checklist 
![image](https://github.com/mate-academy/jv-linked-list/assets/73792778/e9dad4a3-bfd8-4405-b0e3-d6ac438827d2)

Reply:
You're right. Didn't sean that cod is the same.